snmp/datadog_checks/snmp/models.py

```diff
@@ -41,10 +42,7 @@ def __init__(self, value):
         # Resolve the `ObjectIdentity` which we can use to resolve the MIB name of the OID (for metric naming).
         # PySNMP objects may contain MIB information already, so check for them in priority.
         if isinstance(value, ObjectType):
-            # No other choice than to use private API here.
-            object_identity = value._ObjectType__args[0]
-            if not isinstance(object_identity, ObjectIdentity):  # pragma: no cover
-                raise RuntimeError('Expected {!r} to be an `ObjectIdentity` instance'.format(object_identity))
```

Comment:
Removing this check isn't really a behavioral change - I originally added this for assurance purposes during dev, and so far there hasn't been instances of this error. It makes sense, because the only constructor signature for `ObjectType` is `ObjectType(<object_identity>)`.